plymouth: add no echo and show password mode

mati7337 opened the pull request:
Add an option to hide password bullets during disk decryption. A
help message is displayed in top left corner showing which keys
are used to do it.

F1 hides bullets
F2 shows password as bullets

requires https://gitlab.freedesktop.org/plymouth/plymouth/-/commit/b41e40e065c60e76b9721747492875c3454440dc
fixes https://github.com/QubesOS/qubes-issues/issues/2199

![screenshot](https://user-images.githubusercontent.com/99613026/156014225-a21c35e4-868d-444a-a07c-6bdb6f30e9cf.png)

mati7337 committed:
plymouth: add no echo mode

Add an option to hide password bullets during disk decryption. A
help message is displayed in the top left corner showing which keys
are used to switch to it.

F1 hides bullets
F2 shows password as bullets

fixes QubesOS/qubes-issues#2199

mati7337 committed:
plymouth: add show password mode

Add an option to show password as plaintext by pressing F3.

mati7337 commented:
7971a452cd7f92b6f0555954c716b4533c327cdd also adds a show password feature. It requires https://gitlab.freedesktop.org/plymouth/plymouth/-/merge_requests/166 to work.
It fixes the plymouth part of https://github.com/QubesOS/qubes-issues/issues/5521
![screenshot2](https://user-images.githubusercontent.com/99613026/156264362-5a53f014-cb27-48ad-81ca-518687d8f2c7.png)

marmarek commented:
> [7971a45](https://github.com/QubesOS/qubes-artwork/commit/7971a452cd7f92b6f0555954c716b4533c327cdd) also adds a show password feature. It requires https://gitlab.freedesktop.org/plymouth/plymouth/-/merge_requests/166 to work.

I kind of suspect upstream won't be happy about changing API of existing functions - it will break existing themes. But lets see what they say.

marmarek commented on plymouth/qubes-dark/qubes-dark.script:
```diff
@@ -41,6 +49,9 @@ fun dialog_setup(text)
     entry.sprite.SetY(prompt.sprite.GetY() + prompt.image.GetHeight() + 16);
     entry.sprite.SetZ(1);
+
+    help.image = Image.Text("F1 - Hide password bullets\nF2 - Show password as bullets\nF3 - Show password as text", 0.8, 0.8, 0.8);
```
Please make it some further key, for example F8. Otherwise it's too easy to show the password when in fact you want to hide it.

mati7337 replied:
Changing the show password button to F8 and keeping F1 and F2 will be ok? Or should I change them too?

marmarek replied:
Yes, keep F1/F2. The point is to have them further apart.

mati7337 commented:
> > [7971a45](https://github.com/QubesOS/qubes-artwork/commit/7971a452cd7f92b6f0555954c716b4533c327cdd) also adds a show password feature. It requires https://gitlab.freedesktop.org/plymouth/plymouth/-/merge_requests/166 to work.
> 
> I kind of suspect upstream won't be happy about changing API of existing functions - it will break existing themes. But lets see what they say.

I tested it and it shouldn't break existing themes. It just won't pass the extra argument to the callback function.

mati7337 committed:
plymouth: make show password and show password as bullets keys be further apart

Make show password and show password as bullets keys be further appart so it's
harder to press the show password key accidentally

marmarek commented:
Please update for the new version of https://gitlab.freedesktop.org/plymouth/plymouth/-/merge_requests/166

mati7337 committed:
plymouth: update theme to work with 9ee0a053021f478b2f05b7aaa2512edc3c4123ef

The password now gets passed to a DisplayPrompt callback instead
of a DisplayPassword callback

mati7337 commented:
> Please update for the new version of https://gitlab.freedesktop.org/plymouth/plymouth/-/merge_requests/166

Done: [75d7f01e878079fbe6ea88281200cd0c169ddbc7](https://github.com/QubesOS/qubes-artwork/pull/29/commits/75d7f01e878079fbe6ea88281200cd0c169ddbc7)

marmarek reviewed (changes requested):
Two remarks:
 - please make the help text visible only when password prompt is visible.
- in the plaintext password mode, the password remains on the screen even after the prompt is already finalized

mati7337 committed:
plymouth: make help only visible during password entry and hide plaintext password after entry

mati7337 commented:
> Two remarks:
> 
>     * please make the help text visible only when password prompt is visible.
> 
>     * in the plaintext password mode, the password remains on the screen even after the prompt is already finalized

[1569b8a32824e8cadb0c332018f3932b9a45d2db](https://github.com/QubesOS/qubes-artwork/pull/29/commits/1569b8a32824e8cadb0c332018f3932b9a45d2db) should fix it
